### PR TITLE
Allow Buffer side by side with byte[]

### PR DESCRIPTION
**Pull request by pmlopes**
Signed-off-by: Paulo Lopes <pmlopes@gmail.com>

Motivation:

This is an enhancement planned for 4.0 where not only byte[] are allowed but also `Buffer`.

### CODE REVIEW

**Comment by pmlopes** [comment]
See: https://github.com/vert-x3/issues/issues/500

**Comment by pmlopes** [comment]
@vietj added tests to codec

File: src/main/java/io/vertx/core/json/JsonObject.java [hunk 1]
```diff
@@ -285,12 +287,50 @@ public JsonArray getJsonArray(String key) {
...
+      return ((Buffer) val).getBytes();
+    }
     // assume that the value is in String format as per RFC
     String encoded = (String) val;
     // parse to proper type
     return BASE64_DECODER.decode(encoded);
   }
 
+  /**
+   * Get the Buffer value with the specified key.
```
Comment (vietj) [inline src/main/java/io/vertx/core/json/JsonObject.java:301]: {@code key}

File: src/main/java/io/vertx/core/json/JsonObject.java [hunk 2]
```diff
@@ -285,12 +287,50 @@ public JsonArray getJsonArray(String key) {
...
+   *
+   * JSON itself has no notion of a binary, this extension complies to the RFC-7493, so this method assumes there is a
+   * String value with the key and it contains a Base64 encoded binary, which it decodes if found and returns.
+   *
+   * @param key the key to return the value for
+   * @return the value or null if no value for that key
+   * @throws java.lang.ClassCastException       if the value is not a String
```
Comment (vietj) [inline src/main/java/io/vertx/core/json/JsonObject.java:308]: _string_

File: src/main/java/io/vertx/core/json/JsonObject.java [hunk 3]
```diff
@@ -285,12 +287,50 @@ public JsonArray getJsonArray(String key) {
...
+   * @throws java.lang.IllegalArgumentException if the String value is not a legal Base64 encoded value
```
Comment (vietj) [inline src/main/java/io/vertx/core/json/JsonObject.java:309]: simply "value" and not "String value"

**Comment by vietj** [comment]
@pmlopes ping

**Comment by pmlopes** [comment]
@vietj resolved!

**Comment by vietj** [comment]
thanks do you know why the PR complains about a conflict ?

**Comment by vietj** [comment]
@pmlopes can you merge this PR, I cannot do it from web interface because it complains about conflict that I can't see

**Pull request merged**